test-case-reporting/types/test-case-reporting.d.ts

```diff
@@ -102,7 +106,7 @@ declare module 'test-case-reporting' {
     }
 
     export interface TestRun {
-      // See comment under `DB.Build.id`
+      // Nullable because it is not set when uploading. Not nullable in the database.
```

Comment:
I think rather than adding comments in each case, one comment at the top of the `DB` namespace would be enough to explain that some fields will be optional because they are set by the DB automatically

Reply:
Good idea. Done.